File: leetCode/fukun/lc_110.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

var isBalanced = function (root) {
    const calHeight = function (node) {
        if (!node) return 0;
        let ll = calHeight(node.left),
            lr = calHeight(node.right);
        if (ll === false || lr === false) return false;
        if (Math.abs(ll - lr) >= 2) return false;
        return Math.max(ll, lr) + 1;
    };
    return calHeight(root) !== false;
};


let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(2);
console.log(isBalanced(t));

File: leetCode/lyc/e/lc_110.java
/*Balanced Binary Tree*/
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public class Node{
        boolean istrue;
        int depth;
        
        public Node(boolean istrue,int depth){
            this.istrue = istrue;
            this.depth = depth;
        }
    }
    
    public boolean isBalanced(TreeNode root) {
        return findbalance(root).istrue;
    }
    
    public Node findbalance(TreeNode root){
        if(root == null){
            return new Node(true,0);
        }
        
        Node left = findbalance(root.left);
        Node right = findbalance(root.right);
        
        if(left.istrue == false || right.istrue == false){
            return new Node(false,0);
        }
        if(Math.abs(left.depth - right.depth) > 1){
            return new Node(false,0);
        }
        
        return new Node(true,Math.max(left.depth,right.depth) + 1);
    }
}